python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -104,22 +107,45 @@ def _parse_op_fn(op: AbstractUDFMap):
         fn_constructor_args = op._fn_constructor_args or ()
         fn_constructor_kwargs = op._fn_constructor_kwargs or {}
 
-        op_fn = make_callable_class_concurrent(op_fn)
+        if inspect.isasyncgenfunction(op._fn.__call__):
 
-        def fn(item: Any) -> Any:
-            assert ray.data._cached_fn is not None
-            assert ray.data._cached_cls == op_fn
-            try:
-                return ray.data._cached_fn(item, *fn_args, **fn_kwargs)
-            except Exception as e:
-                _handle_debugger_exception(e)
+            def init_fn():
+                if ray.data._cached_fn is None:
+                    ray.data._cached_cls = op_fn
+                    ray.data._cached_fn = op_fn(
+                        *fn_constructor_args, **fn_constructor_kwargs
+                    )
+                    loop = asyncio.new_event_loop()
+                    asyncio.set_event_loop(loop)
+                    ray.data._cached_loop = loop
 
-        def init_fn():
-            if ray.data._cached_fn is None:
-                ray.data._cached_cls = op_fn
-                ray.data._cached_fn = op_fn(
-                    *fn_constructor_args, **fn_constructor_kwargs
-                )
+            async def fn(item: Any) -> Any:
+                assert ray.data._cached_fn is not None
+                assert ray.data._cached_cls == op_fn
+                assert ray.data._cached_loop is not None
```

Comment:
to make code cleaner, let's define a class (e.g. `MapActorContext`) to capture all these variables